src/fu-self-test.c

```diff
@@ -3176,6 +3178,162 @@ _plugin_device_register_cb(FuPlugin *plugin, FuDevice *device, gpointer user_dat
 	fu_plugin_runner_device_register(plugin, device);
 }
 
+static void
+fu_backend_usb_func(gconstpointer user_data)
+{
+	FuTest *self = (FuTest *)user_data;
+	gboolean ret;
+	FuDevice *device_tmp;
+	g_autofree gchar *gusb_emulate_fn = NULL;
+	g_autofree gchar *devicestr = NULL;
+	g_autoptr(FuBackend) backend = fu_usb_backend_new(self->ctx);
+	g_autoptr(FuDeviceLocker) locker = NULL;
+	g_autoptr(FuProgress) progress = fu_progress_new(G_STRLOC);
+	g_autoptr(GError) error = NULL;
+	g_autoptr(GPtrArray) devices = NULL;
+	g_autoptr(GPtrArray) possible_plugins = NULL;
+	g_autoptr(JsonParser) parser = json_parser_new();
+
+#if !G_USB_CHECK_VERSION(0, 4, 0)
+	g_test_skip("GUsb version too old");
+	return;
+#endif
+
+	/* generate the fake base64 chunks in the usb-devices.json file */
+	if (g_getenv("FWUPD_GENERATE_DS20") != NULL) {
```

Comment:
why is this gated by the env variable?

Reply:
I think all this can do away if/when we have the *generate-ds20-blob.py* script.

Reply:
I've removed this whole chunk thanks to `fwupdtool firmware-build`